Add documentation on ReadTheDocs

turnerm opened the pull request:
In this PR I've tried to improve / extend the documentation, and get it up on [ReadTheDocs](https://aa-toolbox.readthedocs.io/en/latest/) (I currently have it pointing to this branch; it should be set to `main` upon release). 

This should address #7, apart from adding some info to PyPI.

Tinkaa reviewed (commented):
Thanks so much for all this work! Left some small comments. Also felt free to improve the IRI background a little as that was my own writing. 

Few other thoughts: 
- Do you think we can pimp the layout a bit? Make it look a bit less 90s and e.g. add our logo? 
- Should we explain somewhere the 3-step process we follow of download, process, load? 
- Sometimes the language of (setting-up) things felt quite advanced to me, even having groups like 510 in mind. Added some suggestions and also don't know how to make it even more clear (and how basic we want to have it), but just wanted to drop it here. 
- How much background info on data sources do we want to give? E.g. for IRI do we also want to explain what a tercile forecast is?

turnerm commented on CONTRIBUTING.rst:
````diff
@@ -0,0 +1,114 @@
+``aa-toolbox`` makes use of
+```geopandas`` <https://geopandas.org/en/stable/>`__, which depends on
+```Fiona`` <https://github.com/Toblerity/Fiona>`__, so you will need to
+have ```GDAL`` <https://github.com/Toblerity/Fiona#installation>`__
````
Ah oops, I used `pandoc` to convert and indeed I think that caused the issue

turnerm commented on docs/source/api.rst:
```diff
@@ -0,0 +1,5 @@
+AA Toolbox API
```
Hm, so this is supposed to autogenerate the API docs from the docstrings. It works when I run it on my machine but not RTD, I will need to look into it

Tinkaa replied:
I dont know if you want to leave this for another PR or it is working for you. But just wanted to flag it is still not working for me online (though it is working locally for me as well).

turnerm replied:
Yes I'm working on it for this PR, hence all the commits lately :zany_face:

turnerm commented on docs/source/datasources/codab.rst:
```diff
@@ -0,0 +1,58 @@
+    country_config = create_country_config(iso3="npl")
+    codab = CodAB(country_config=country_config)
+    codab.download()
+    npl_admin1 = codab.load(admin_level=1)
```
Great points, done!

turnerm commented on docs/source/datasources/iri.rst:
```diff
@@ -0,0 +1,103 @@
+IRI
```
Good point, I will rename the docs. I think we can keep the folder name as we may put additional IRI sources in there, the module itself is called `_seasonal_forecast` and the class names contain `Forecast` so I think it's clear.

turnerm commented on docs/source/index.rst:
```diff
@@ -1,21 +1,46 @@
 
-Welcome to aatoolbox's documentation!
-=====================================
+Welcome to AA Toolbox's documentation. Get set up
```
Good idea, done!

turnerm commented on docs/source/index.rst:
```diff
@@ -1,21 +1,46 @@
+User Guide
+----------
+
+This section includes background information on the AA Toolbox,
```
Great suggestion!

turnerm commented on docs/source/index.rst:
```diff
@@ -1,21 +1,46 @@
+
+   introduction
+   quickstart
+   datasources/codab
```
Good point, I think the only way to do so is to add a separate data sources page, let me know if you think it's okay. The source names don't show up in the side menu though, I'd have to change the overall depth and I'm afraid it would get too messy

Tinkaa replied:
Yes, looks good! I am thinking if a depth of 2 would still be okay, but lets leave it for now.

turnerm commented on docs/source/introduction.rst:
```diff
@@ -0,0 +1,19 @@
+Introduction
+============
+
+`Anticipatory action <https://centre.humdata.org/anticipatory-action/>`_
```
Agree that it's nice, but do you know an easy way to do this with restructured text?

Tinkaa replied:
Ah you are 100% right. Apparently rst is not supposed to that. Nvm then :)

turnerm commented on docs/source/introduction.rst:
```diff
@@ -0,0 +1,19 @@
+some of our own AA frameworks. See the links below for more info:
+
+* `Drought in Burkina Faso <https://github.com/OCHA-DAP/pa-aa-bfa-drought>`_
+* `Cholera in DRC <https://github.com/OCHA-DAP/pa-aa-cod-infectious-disease>`_
```
GOod point, I've added in to contact us for more info. 
 
Side note: I've just been using centrehumdata@un.org as the contact email. Do you know if the PA team has one, should we?

Tinkaa replied:
Yeah I have asked this once and I dont think we have another one. As it seems Leonardo normally gets the requests forwarded from that email, I think it is fine for now. If we get a lot of traction, we could consider making a separate one.

turnerm commented on docs/source/quickstart.rst:
```diff
@@ -0,0 +1,64 @@
+    $ pip install aa-toolbox
+
+AA Toolbox should now be installed.
+
```
I'm not sure if I would include it in quickstart because I would consider it more advanced usage. I did add some config docs in the custom config PR which I'm about to merge in, let me know if you think it's enough.

Tinkaa replied:
Yes, looks great

turnerm commented:
Thank you so much Tinka for the very helpful comments, I think they really improved the clarity! 

> Do you think we can pimp the layout a bit? Make it look a bit less 90s and e.g. add our logo?

Aha, ouch! I thought this is the nicest one of the [built-ins](https://www.sphinx-doc.org/en/master/usage/theming.html), also because it's the closest to [Flask](https://flask.palletsprojects.com/en/2.1.x/). Maybe we can take one from [here](https://sphinx-themes.org/#themes), is there one that you like? 

> Should we explain somewhere the 3-step process we follow of download, process, load?

I've added this to `datasource.rst`. 

> Sometimes the language of (setting-up) things felt quite advanced to me, even having groups like 510 in mind. Added some suggestions and also don't know how to make it even more clear (and how basic we want to have it), but just wanted to drop it here.

Yeah for sure, maybe we can eventually ask for e.g. Caitlyn's help. 

> How much background info on data sources do we want to give? E.g. for IRI do we also want to explain what a tercile forecast is?

I would say while of course more is better, we shouldn't put the cart before the horse and spend lots of time writing these docs that nobody uses (yet!). If it's clear that it will be useful to people then definitely add it!

Tinkaa commented:
Looks great @turnerm! 

>Aha, ouch! I thought this is the nicest one of the [built-ins](https://www.sphinx-doc.org/en/master/usage/theming.html), also because it's the closest to [Flask](https://flask.palletsprojects.com/en/2.1.x/). Maybe we can take one from [here](https://sphinx-themes.org/#themes), is there one that you like?

Hahahahah. But, I am confused. Are we using the "sphinx_rtd_theme" as I see that in the `conf.py`? Cause to be honest I do think the Flask docs are ugly. But I do like the template of [sphinx_rtd_theme](https://sphinx-rtd-theme.readthedocs.io/en/stable/) and e.g. its use for [the Altair library](https://altair-viz.github.io/)

Tinkaa commented:
Fixed a FN code block bug I made.

turnerm commented:
> Cause to be honest I do think the Flask docs are ugly

:joy: oh wow shots fired @Tinkaa. I switched back to RTD theme, it perhaps does look cleaner and I trust your judgement on this one! 

Oh and last thing, you mentioned about a logo. WHich logo did you think we should add?
